### PR TITLE
chore(E2E) : Configuring a single metric widget

### PR DESCRIPTION
**Pull request by MedWassCh**
## Description
Implemented tests from dashboards' single metric widget automated test battery

**Fixes** # (issue)
MON-23784 :Creating and configuring a new Single Metric widget on a dashboard 
MON-23786 :Duplicating a Single metric widget
MON-23791:Editing the value format of a single metric widget
MON-23789:Editing the thresholds of a Single metric widget
MON-23787 :Editing the display type of a Single metric widget
 MON-23799: Deleting a Single metric widget
## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/tests/e2e/fixtures/dashboards/creation/widgets/dashboadWithDoubleWidget.json
Comment (paul-oureib) [inline centreon/tests/e2e/fixtures/dashboards/creation/widgets/dashboadWithDoubleWidget.json:1]: rename it "dashboardWithTwoWidgets" (+ correct missing r)

File: centreon/tests/e2e/fixtures/dashboards/creation/widgets/dashboadWithDoubleWidget.json [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+                           "has_passive_checks_enabled":false,
+                           "last_status_change":"2023-11-07T09:22:46+01:00",
+                           "tries":"1/3 (H)",
+                           "information":"OK - 127.0.0.1 rta 0.019ms lost 0%",
+                           "performance_data":null,
+                           "is_notification_enabled":false,
+                           "severity":null,
+                           "links":{
+                              "endpoints":{
+                                 "details":"/centreon/api/latest/monitoring/resources/hosts/14/services/26",
```
Comment (paul-oureib) [inline centreon/tests/e2e/fixtures/dashboards/creation/widgets/dashboadWithDoubleWidget.json:153]: are these numbers (14, 26) consistent? shouldn't you parametrize them one way or another?

File: centreon/tests/e2e/features/Dashboards/09-widget-single-metric-configuration/index.ts [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+);
+
+When(
+  'the dashboard administrator user selects a resource and the metric for the widget to report on',
+  () => {
+    cy.getByLabel({ label: 'Title' }).type(genericTextWidgets.default.title);
+    cy.getByLabel({ label: 'RichTextEditor' })
+      .eq(0)
+      .type(genericTextWidgets.default.description);
+    cy.get('[class*="MuiSelect-select"]').click();
```
Comment (Thebarda) [inline centreon/tests/e2e/features/Dashboards/09-widget-single-metric-configuration/index.ts:117]: Prefer label text or data testid instead of classnames

**Review by arnapou — APPROVED** [review]

**Review by arnapou — APPROVED** [review]

File: centreon/tests/e2e/support/commands.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 /* eslint-disable @typescript-eslint/no-namespace */
 
 import 'cypress-wait-until';
+import 'cypress-real-events/support';
```
Comment (kduret) [inline centreon/tests/e2e/support/commands.ts:4]:
```suggestion
```

**Review by kduret — DISMISSED** [review]

**Review by paul-oureib — DISMISSED** [review]

**Review by arnapou — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Review by paul-oureib — APPROVED** [review]

**Pull request merged**